vector/src/main/java/im/vector/app/features/invite/InviteUsersToRoomViewModel.kt

```diff
@@ -55,39 +56,38 @@ class InviteUsersToRoomViewModel @AssistedInject constructor(
+        _viewEvents.post(InviteUsersToRoomViewEvents.Loading)
+        selections.asFlow()
+                .map { user ->
+                    when (user) {
+                        is PendingSelection.UserPendingSelection     -> room.membershipService().invite(user.user.userId, null)
+                        is PendingSelection.ThreePidPendingSelection -> room.membershipService().invite3pid(user.threePid)
                     }
-                    .catch { cause ->
-                        _viewEvents.post(InviteUsersToRoomViewEvents.Failure(cause))
+                }
+                .catch { cause ->
+                    _viewEvents.post(InviteUsersToRoomViewEvents.Failure(cause))
+                }
+                .onCompletion {
+                    val successMessage = when (selections.size) {
+                        1    -> stringProvider.getString(
+                                R.string.invitation_sent_to_one_user,
+                                selections.first().getBestName()
+                        )
+                        2    -> stringProvider.getString(
+                                R.string.invitations_sent_to_two_users,
+                                selections.first().getBestName(),
+                                selections.last().getBestName()
+                        )
+                        else -> stringProvider.getQuantityString(
+                                R.plurals.invitations_sent_to_one_and_more_users,
+                                selections.size - 1,
+                                selections.first().getBestName(),
+                                selections.size - 1
+                        )
                     }
-                    .collect {
-                        val successMessage = when (selections.size) {
-                            1    -> stringProvider.getString(
-                                    R.string.invitation_sent_to_one_user,
-                                    selections.first().getBestName()
-                            )
-                            2    -> stringProvider.getString(
-                                    R.string.invitations_sent_to_two_users,
-                                    selections.first().getBestName(),
-                                    selections.last().getBestName()
-                            )
-                            else -> stringProvider.getQuantityString(
-                                    R.plurals.invitations_sent_to_one_and_more_users,
-                                    selections.size - 1,
-                                    selections.first().getBestName(),
-                                    selections.size - 1
-                            )
-                        }
-                        _viewEvents.post(InviteUsersToRoomViewEvents.Success(successMessage))
-                    }
-        }
+                    _viewEvents.post(InviteUsersToRoomViewEvents.Success(successMessage))
```

Comment:
The ViewEvent was posted too early and so the Activity is finished, the coroutine scope is cancelled, and the pending invitation request are cancelled too.